app/src/main/java/com/eis/audioUtility/AudioUtilityManager.java

```diff
@@ -0,0 +1,199 @@
+     */
+    private static int getStream(AUMStream stream) throws IllegalArgumentException {
+
+        switch (stream) {
+            case ALARM:
+                return AudioManager.STREAM_ALARM;
+            case RING:
+                return AudioManager.STREAM_RING;
+            case MUSIC:
+                return AudioManager.STREAM_MUSIC;
+            default:
+                throw new IllegalArgumentException("\nIllegal Stream. It should be ALARM, RING or MUSIC .");
+        }
+    }
+
+    /**
+     * @param context The current Context.
+     * @return The current Vibrator instance.
+     */
+    protected static Vibrator getVibrator(Context context) {
+        return (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
+    }
+
+    /**
+     * Makes the target device vibrate.
+     *
+     * @param context The current Context.
+     */
+    public static void singleVibrate(Context context) {
+        int time = VIBRATION_TIME + VIBRATION_DELAY;
+        // Check if API version is >26 or not
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getVibrator(context).vibrate(VibrationEffect.createOneShot(time, VibrationEffect.DEFAULT_AMPLITUDE));
+        } else {
+            //deprecated in API 26
+            getVibrator(context).vibrate(time);
+        }
+    }
+
+    /**
+     * Makes the target device vibrate multiple times, with default pattern
+     *
+     * @param context The current Context.
+     */
+    public static void multipleVibrate(Context context) {
+        // Vibration pattern
+        long[] pattern = {VIBRATION_DELAY, VIBRATION_TIME, VIBRATION_DELAY, VIBRATION_TIME};
+        // Check if API version is >26 or not
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getVibrator(context).vibrate(VibrationEffect.createWaveform(pattern, VIBRATION_REPEAT_CODE));
+        } else {
+            //deprecated in API 26
+            getVibrator(context).vibrate(pattern, VIBRATION_REPEAT_CODE);
+        }
+    }
+
+    /**
+     * Makes the target device vibrate multiple times, with a certain pattern
+     *
+     * @param context The current Context.
+     * @param pattern The pattern of the vibration, created in couples of pause-vibration.
+     */
+    public static void multipleVibrate(Context context, long[] pattern) {
+        // Check if API version is >26 or not
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            getVibrator(context).vibrate(VibrationEffect.createWaveform(pattern, VIBRATION_REPEAT_CODE));
+        } else {
+            //deprecated in API 26
+            getVibrator(context).vibrate(pattern, VIBRATION_REPEAT_CODE);
+        }
+    }
+
+    /**
+     * This method stops the vibration.
+     *
+     * @param context The current Context.
+     */
+    public static void stopVibrate(Context context) {
+        Vibrator vibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
```

Comment:
Should have used the `getVibrator()` method

Reply:
Oh, you are totally right!
Thank you for pointing this out.